vultr/loadbalancers.go

```diff
@@ -571,17 +573,41 @@ func buildForwardingRules(service *v1.Service) ([]govultr.ForwardingRule, error)
 
 	for _, port := range service.Spec.Ports {
 		// default the port
-		protocol := defaultProtocol
+		frontendProtocol := defaultProtocol
+		backendProtocol := getBackendProtocol(service)
 
 		if httpsPorts[port.Port] {
 			if getSSLPassthrough(service) {
-				protocol = protocolTCP
+				frontendProtocol = protocolTCP
 			} else {
-				protocol = protocolHTTPs
+				frontendProtocol = protocolHTTPs
 			}
 		}
 
-		rule, err := buildForwardingRule(&port, protocol)
+		// Check frontend/backend port combinations (listed below what is acceptable)
+		// frontend = tcp: backend must be tcp
+		// frontend = https: backend can be http(s)
+		// frontend = http: backend can be http(s)
+		switch frontendProtocol {
+		case "tcp":
+			if backendProtocol != "tcp" {
+				klog.V(3).Infof("When frontend proto is tcp, backend must also be tcp, setting backend to tcp")
+				backendProtocol = "tcp"
+			}
+		case "http", "https":
+			if backendProtocol != "http" && backendProtocol != "https" {
```

Comment:
We should not default to HTTPS for HTTP frontends, only HTTPS frontends. 
Defaults should be:
HTTP -> HTTP
HTTPS -> HTTPS
TCP -> TCP

Reply:
Fixed in 021f41a